R/primary-coding-suep.R

```diff
@@ -637,52 +660,53 @@ build_who_scale_suep_df <- function(trial_data, pid, docid, visitid){
-                                                 .data$is_hospital & .data$ecu_main_diag_covid == "Hauptdiagnose Andere" ~ "Hospitalisiert mit Covid, moderate Phase"),
-      ecu_who_scale_with_diag = as.integer(case_when(.data$ecu_who_scale_with_diag.factor == "Keine Informationen verfuegbar" ~ -1,
+      ecu_who_scale.factor = case_when(str_detect(.data$ecu_pr_inclusion, "Kontroll") ~ "Kontrollgruppe, ohne Sars-Infektion",
+                                       .data$ward.factor == "Nein" ~ "Ambulant, milde Phase", # ward == 0 = "Nein" 
+                                       .data$ward.factor == "Keine Informationen verf\u00fcgbar" | 
+                                         .data$gec_oxy.factor == "Keine Informationen verf\u00fcgbar" ~ "Keine Informationen verf\u00fcgbar",
+                                       .data$is_hospital == 0 ~ "Ambulant, milde Phase",
+                                       .data$visit_date >= .data$ecu_death_date.date ~ "Verstorben",
+                                       .data$is_oxy_type_severe & .data$is_hospital == 1 ~ "Hospitalisiert, schwere Phase",
+                                       .data$is_hospital == 1  & .data$is_oxy != "-1" ~ "Hospitalisiert, moderate Phase"),
+      ecu_who_scale = as.integer(case_when(.data$ecu_who_scale.factor == "Keine Informationen verf\u00fcgbar" ~ -1,
+                                           .data$ecu_who_scale.factor == "Kontrollgruppe, ohne Sars-Infektion" ~ 0,
+                                           .data$ecu_who_scale.factor == "Ambulant, milde Phase" ~ 1,
+                                           .data$ecu_who_scale.factor == "Hospitalisiert, moderate Phase" ~ 2,
+                                           .data$ecu_who_scale.factor == "Hospitalisiert, schwere Phase" ~ 3,
+                                           .data$ecu_who_scale.factor == "Verstorben" ~ 4)),
+      ecu_who_scale_with_diag.factor = case_when(ecu_who_scale.factor == "Hospitalisiert, moderate Phase" & .data$ecu_main_diag_covid == "Hauptdiagnose Covid" ~ "Hospitalisiert wegen Covid, moderate Phase",
+                                                 ecu_who_scale.factor == "Hospitalisiert, moderate Phase" & .data$ecu_main_diag_covid == "Hauptdiagnose Andere" ~ "Hospitalisiert mit Covid, moderate Phase",
+                                                 ecu_who_scale.factor == "Hospitalisiert, schwere Phase" & .data$ecu_main_diag_covid == "Hauptdiagnose Covid" ~ "Hospitalisiert wegen Covid, schwere Phase",
+                                                 ecu_who_scale.factor == "Hospitalisiert, schwere Phase" & .data$ecu_main_diag_covid == "Hauptdiagnose Andere" ~ "Hospitalisiert mit Covid, schwere Phase",
+                                                 is.na(.data$ecu_main_diag_covid) | .data$ecu_main_diag_covid == "Keine Informationen verf\u00fcgbar" ~ "Keine Informationen verf\u00fcgbar",
```

Comment:
i changed it so that is.na(.data$ecu_main_diag_covid) returns NA and .data$ecu_main_diag_covid == "Keine Informationen verf\u00fcgbar" returns "Keine Informationen verf\u00fcgbar"